src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java

```diff
@@ -104,55 +104,135 @@
+ *        break;
+ *    }
+ *  }
+ * }
  * </pre>
  * <p>
- * To configure the check to produce a violation on comparing {@code this} with
- * {@code null}(i.e. {@code this == null} and {@code this != null}):
+ * To configure the check to produce a violation on a switch with too many cases:
  * </p>
  * <pre>
  * &lt;module name=&quot;DescendantToken&quot;&gt;
- *   &lt;property name=&quot;tokens&quot; value=&quot;EQUAL,NOT_EQUAL&quot;/&gt;
- *   &lt;property name=&quot;limitedTokens&quot; value=&quot;LITERAL_THIS,LITERAL_NULL&quot;/&gt;
- *   &lt;property name=&quot;maximumNumber&quot; value=&quot;1&quot;/&gt;
- *   &lt;property name=&quot;maximumDepth&quot; value=&quot;1&quot;/&gt;
- *   &lt;property name=&quot;sumTokenCounts&quot; value=&quot;true&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;LITERAL_SWITCH&quot;/&gt;
+ *   &lt;property name=&quot;limitedTokens&quot; value=&quot;LITERAL_CASE&quot;/&gt;
+ *   &lt;property name=&quot;maximumDepth&quot; value=&quot;2&quot;/&gt;
+ *   &lt;property name=&quot;maximumNumber&quot; value=&quot;2&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
- * To configure the check to produce a violation on a {@code String} literal equality check:
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *   public void foo() {
+ *     int x = 1;
+ *     switch (x) { // ok
+ *       case 1:
+ *         // Some code
+ *         break;
+ *       default:
+ *         // Some code
+ *         break;
+ *     }
+ *
+ *     switch (x) { // violation
+ *       case 1:
+ *         // Some code
+ *         break;
+ *       case 2:
+ *         // Some code
+ *         break;
+ *       default:
+ *         // Some code
+ *         break;
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to produce a violation on a switch that is nested in another switch:
  * </p>
  * <pre>
  * &lt;module name=&quot;DescendantToken&quot;&gt;
- *   &lt;property name=&quot;tokens&quot; value=&quot;EQUAL,NOT_EQUAL&quot;/&gt;
- *   &lt;property name=&quot;limitedTokens&quot; value=&quot;STRING_LITERAL&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;LITERAL_SWITCH&quot;/&gt;
+ *   &lt;property name=&quot;limitedTokens&quot; value=&quot;LITERAL_SWITCH&quot;/&gt;
  *   &lt;property name=&quot;maximumNumber&quot; value=&quot;0&quot;/&gt;
- *   &lt;property name=&quot;maximumDepth&quot; value=&quot;1&quot;/&gt;
+ *   &lt;property name=&quot;minimumDepth&quot; value=&quot;1&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
- * To configure the check to produce a violation on an assert statement that may
- * have side effects (formatted for browser display):
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *   public void foo() {
+ *     int x = 1;
+ *     int y = 2;
+ *     switch (x) { // ok
+ *       case 1:
+ *         System.out.println("xyz");
+ *         break;
+ *     }
+ *     switch (y) { // ok
+ *       case 1:
+ *         switch (y) { // violation
```

Comment:
Shouldn't violation be placed on outer switch?

Reply:
Yes, it should be present on outer switch.

Reply:
Done